configure.py

```diff
@@ -2402,5 +2403,56 @@ def create_building_system(args):
     generate_compdb('compile_commands.json', ninja, args.buildfile, selected_modes)
 
 
+def configure_for_mode(mode, args):
+    # TODO: use Ninja multi-config
```

Comment:
Isn't this done below?

Reply:
no. multi-config is different. currently, we generate <n> build files for each of the build modes. while [Multi-Config](https://cmake.org/cmake/help/latest/generator/Ninja%20Multi-Config.html) creates single build file. it's quite like the same way how we create the monolithic `build.ninja` for all modes using `configure.py`. apparently, the mult-config can save us some CPU cycle and disk space, as we don't need to build the same set of artifacts which are identical between different build modes. for instance the artifacts built from submodules under `tools`.